src/Adapter/CatalogPriceRule/CommandHandler/EditCatalogPriceRuleHandler.php

```diff
@@ -0,0 +1,122 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\CatalogPriceRule\CommandHandler;
+
+use PrestaShop\PrestaShop\Core\Domain\CatalogPriceRule\Command\EditCatalogPriceRuleCommand;
+use PrestaShop\PrestaShop\Core\Domain\CatalogPriceRule\CommandHandler\EditCatalogPriceRuleHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\CatalogPriceRule\Exception\CatalogPriceRuleException;
+use PrestaShop\PrestaShop\Core\Domain\CatalogPriceRule\Exception\UpdateCatalogPriceRuleException;
+use PrestaShopException;
+use SpecificPriceRule;
+
+/**
+ * Handles command which edits catalog price rule handler using legacy object model
+ */
+final class EditCatalogPriceRuleHandler implements EditCatalogPriceRuleHandlerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(EditCatalogPriceRuleCommand $command)
+    {
+        try {
+            $specificPriceRule = $this->createSpecificPriceRuleFromCommand($command);
+
+            if (false === $specificPriceRule->validateFields(false)) {
+                throw new CatalogPriceRuleException('Specific price rule contains invalid field values');
+            }
+
+            if (false === $specificPriceRule->update()) {
+                throw new UpdateCatalogPriceRuleException(
+                    sprintf('Failed to update specific price rule with id %s', $specificPriceRule->id)
+                );
+            }
+            $specificPriceRule->deleteConditions();
```

Comment:
So we need to call this everytime we edit it ?

Reply:
This should be checked more in-depth on separate PR of condition groups. Added a reminder here https://github.com/PrestaShop/PrestaShop/issues/10549#issuecomment-513218354